cmd/silence/main.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/user"
+	"time"
+
+	"bytes"
+
+	"bosun.org/util"
+)
+
+var (
+	flagHost     = flag.String("h", "bosun", "Hostname of your bosun server, defaults to bosun.")
+	flagUser     = flag.String("u", "", "Username, defaults to the username returned from the OS.")
+	flagDuration = flag.String("d", "30m", "A duration to silence this host for. Defaults to 30m and the format is defined at http://golang.org/pkg/time/#ParseDuration")
+	flagTags     = flag.String("t", "", "OpenTSDB tags to be silenced in the format of tagKey=value,tagKey=value. Defaults to host=<hostname>.")
+	flagAlert    = flag.String("a", "", "Name of the alert to silence, defaults to empty which means all alerts.")
+	flagMessage  = flag.String("m", "", "Reason for the silence, defaults to an empty string.")
+	flagForget   = flag.String("f", "", "Set to 'true' to forget anything that goes unknown during the silence. Used when decommissioning something.")
+)
+
+func main() {
+	flag.Parse()
+	u, err := user.Current()
+	if err != nil {
+		log.Fatal(err)
+	}
+	un := u.Username
+	sudo := os.Getenv("SUDO_USER")
+	if sudo != "" {
+		un = sudo
+	}
+	if *flagTags == "" {
+		*flagTags = "host=" + util.Hostname
+	}
+	now := time.Now().UTC()
+	d, err := time.ParseDuration(*flagDuration)
+	if err != nil {
+		log.Fatal(err)
+	}
+	end := now.Add(d)
+	if *flagForget != "" {
+		*flagForget = "true"
+	}
+	s := struct {
```

Comment:
It is not, the API to create silences differs a fair amount from the silence struct itself, I started to change this, but then with all the date parsing etc I decided it wasn't worth the refactor